.github/workflows/sbom-generator.yaml

```diff
@@ -23,7 +26,30 @@ jobs:
         id: sbom
         env: 
           GITHUB_TOKEN: ${{ github.token }}
+
       - uses: actions/upload-artifact@v3.1.0
         with: 
           path: ${{steps.sbom.outputs.fileName }}
           name: "SBOM"
+
+      - name: Get latest release
+        id: latest_release
+        uses: actions/github-script@v4
+        with:
+          script: |
+            const { data } = await github.repos.listReleases({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              per_page: 1
+            });
+            return data[0];
```

Comment:
You're not interested in the latest, what you want is the one that is being triggered,

you can identify this: `if: github.ref_type == tags` or if you trigger on release `if: github.event == release` then the property `github.ref_name` should give you the release you are interested in.